samcli/lib/providers/sam_function_provider.py

```diff
@@ -198,20 +207,27 @@ def _convert_lambda_function_resource(name, resource_properties, layers):  # pyl
         # CodeUri is set to "." in order to get code locally from current directory. AWS::Lambda::Function's ``Code``
         # property does not support specifying a local path
         codeuri = SamFunctionProvider.DEFAULT_CODEURI
+        inlinecode = None
         imageuri = None
         packagetype = resource_properties.get("PackageType", ZIP)
-
         if packagetype == ZIP:
-            codeuri = SamFunctionProvider._extract_lambda_function_code(resource_properties, "Code")
-            LOG.debug("Found Lambda function with name='%s' and CodeUri='%s'", name, codeuri)
+            if "ZipFile" in resource_properties["Code"] and resource_properties["Code"]["ZipFile"] is not None:
+                inlinecode = resource_properties["Code"]["ZipFile"]
+                LOG.debug("Found Lambda function with name='%s' and Code ZipFile", name)
+                codeuri = None
+            else:
+                codeuri = SamFunctionProvider._extract_lambda_function_code(resource_properties, "Code")
+                LOG.debug("Found Lambda function with name='%s' and CodeUri='%s'", name, codeuri)
         elif packagetype == IMAGE:
             imageuri = SamFunctionProvider._extract_lambda_function_imageuri(resource_properties, "Code")
             LOG.debug("Found Lambda function with name='%s' and Imageuri='%s'", name, imageuri)
 
-        return SamFunctionProvider._build_function_configuration(name, codeuri, resource_properties, layers, imageuri)
+        return SamFunctionProvider._build_function_configuration(
+            name, codeuri, resource_properties, layers, inlinecode, imageuri
+        )
 
     @staticmethod
-    def _build_function_configuration(name, codeuri, resource_properties, layers, imageuri=None):
+    def _build_function_configuration(name, codeuri, resource_properties, layers, inlinecode=None, imageuri=None):
```

Comment:
If `inlinecode` is optional, should `codeuri` be optional as well?

Not sure why we are inconsistent with our naming here but I suggest to us `incline_code` rather than `inclinecode`. It is easier to read with underscores.

Reply:
@jfuss this was named without underscore to match the other property names in provider.py

Reply:
Yeah I also agree we should keep these two consistent. We can in fact remove all "=None" from the definition because there is no use of `_build_function_configuration()` that do not provide all arguments.
```suggestion
    def _build_function_configuration(name, codeuri, resource_properties, layers, inlinecode, imageuri):
```

And in the future (if you don't want to do it now), we can add types to them to indicate whether they are optional or not:

```py
def _build_function_configuration(name: str, codeuri: Optional[str], resource_properties: Dict, layers: List[<sometype>], inlinecode: Optional[str], imageuri: Optional[str]):
```

Reply:
Changed it to behave the same way for all three fields